app/repository/RestaurantRepository.java

```diff
@@ -2,10 +2,39 @@
 
 import com.google.inject.ImplementedBy;
 import models.Restaurant;
+import models.RestaurantTable;
+import org.hibernate.Criteria;
+import org.hibernate.Session;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.jpa.HibernateEntityManager;
+import play.db.jpa.JPA;
+
+import java.util.List;
 
 /**
  * Created by lapa on 5/6/17.
  */
-@ImplementedBy(RestaurantRepositoryImplementation.class)
-public interface RestaurantRepository extends Repository<Restaurant> {
+//@ImplementedBy(RestaurantRepositoryImplementation.class)
+public class RestaurantRepository extends RepositoryImplementation<Restaurant> {
+
+        public static Criteria getCriteria(){
+
+        Session session = ((HibernateEntityManager) JPA.em()).getSession();
+        return session.createCriteria(Restaurant.class);
+    }
+
+
+    public Restaurant findRestaurantById(int id) {
```

Comment:
This should be implemented as `findById` method in your base repository implementation so you dont have to implement it for each different model.

Also `id` should be `Long` not `int`.